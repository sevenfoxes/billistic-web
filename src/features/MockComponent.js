import { Component } from "react"

export class MockComponent extends Component {
  render() {
    return null
  }
}
