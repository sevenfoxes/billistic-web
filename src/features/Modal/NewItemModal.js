import React from "react"
import "./NewItem.scss"
import { withStyles } from '@material-ui/core/styles'
import Typography from '@material-ui/core/Typography'
import { compose, withStateHandlers } from "recompose"
import FormControl from '@material-ui/core/FormControl'
import Select from '@material-ui/core/Select'
import InputLabel from '@material-ui/core/InputLabel'
import MenuItem from '@material-ui/core/MenuItem'

const styles = ({ spacing, palette, shadows }) => ({
  paper: {
    width: spacing.unit * 50,
    backgroundColor: palette.background.paper,
    boxShadow: shadows[5],
    padding: spacing.unit * 4,
  },
})

export const NewItemModal = ({ classes, itemType, updateItemType }) => (
  <div tabIndex="-1" className={`modal ${classes.paper}`}>
    <Typography variant="h5" component="h2">
      Add New Item
    </Typography>
    <br />
    <form className="form form_new_item">
      <FormControl className={classes.formControl}>
        <InputLabel htmlFor="age-simple">Item Type</InputLabel>
        <Select
          value={itemType}
          onChange={updateItemType}
          inputProps={{
            name: 'type',
            id: 'income-expense',
          }}
        >
          <MenuItem value='expense'>Expense</MenuItem>
        </Select>
      </FormControl>
    </form>
  </div>
)

export const enhance = compose(
  withStateHandlers(
    ({ initialItemType = 'expense' }) => ({ itemType: initialItemType }),
    {
      updateItemType: () => ({ target }) => ({ itemType: target.value }),
    })
)

export default compose(
  withStyles(styles),
  enhance,
)(NewItemModal)
