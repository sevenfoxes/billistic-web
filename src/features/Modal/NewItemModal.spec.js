import React from "react"
import ShallowRenderer from "react-test-renderer/shallow"
import NewItemModal, { enhance } from './NewItemModal'
import { MockComponent } from '../MockComponent'

import Enzyme, { mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

Enzyme.configure({ adapter: new Adapter() })


describe('NewItemModal', () => {
  it('should match the snapshot', async () => {
    const tree = new ShallowRenderer().render(<NewItemModal />)
    expect(tree).toMatchSnapshot()
  })
})

describe('enhanced component', () => {
  it('should have all the enhancements', async () => {
    const Enhanced = enhance(MockComponent)
    const wrapper = mount(<Enhanced />)
    const mockEvent = {
      target: {
        value: 'some crap',
      },
    }
    let subject = wrapper.find(MockComponent)

    expect(subject.prop('itemType')).toEqual('expense')
    expect(typeof subject.prop('updateItemType')).toEqual('function')

    subject.prop('updateItemType')(mockEvent)
    wrapper.update()
    subject = wrapper.find(MockComponent)

    expect(subject.prop('itemType')).toEqual(mockEvent.target.value)
  })
})
