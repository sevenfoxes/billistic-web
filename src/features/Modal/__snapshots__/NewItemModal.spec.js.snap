// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewItemModal should match the snapshot 1`] = `
<withStateHandlers(NewItemModal)
  classes={
    Object {
      "paper": "withStateHandlers-NewItemModal--paper-1",
    }
  }
/>
`;
